playbooks/upload_to_swift.yml

```diff
@@ -66,7 +66,12 @@
         region_name: "{{ region }}"
       no_log: true
       register: _auth
-      until: (_auth | success) and (auth_token is defined) and (service_catalog is defined)
+      until: (_auth | success) and
```

Comment:
I think you could get rid of all the `and` and brackets and use a list which may be a little easier on the eye.
```yaml
until:
  - _auth | success
  - auth_token | default(none) is not none
  - auth_token | trim != ''
  - service_catalog | default(none) is not none
```

Reply:
I thought so too, but when I tried that yesterday it failed.